implants/lib/transport/src/grpc.rs

```diff
@@ -249,7 +275,7 @@ impl GRPC {
         let mut req = request.into_request();
         req.extensions_mut()
             .insert(GrpcMethod::new("c2.C2", "ReportCredential"));
-        self.grpc.unary(req, path, codec).await
+        self.grpc.as_mut().unwrap().unary(req, path, codec).await
```

Comment:
Unwrap!

Reply:
yeah we check to see if the value exists above each of these so i think it should be okay? it seems extremely difficult to have it be None'd out in between the check and when we use it. that being said it is a TOCTOU